docs/MPURegionsUsageGuide.md

```diff
@@ -0,0 +1,286 @@
+# Developer guide for Usage of MPU Regions in TizenRT
+
+## CONTENTS
+- [Purpose of MPU](#purpose-of-mpu)
+- [Details of MPU Configuration and Attributes](#details-of-mpu-configuration-and-attributes)
+- [Prerequisites for MPU in TizenRT](#prerequisites-for-mpu-in-tizenrt)
+- [Functions used and Enumeration for MPU regions](#functions-used-and-enumeration-for-mpu-regions)
+- [Current Usage of MPU Regions in TizenRT](#current-usage-of-mpu-regions-in-tizenrt)
+- [How to Add new MPU regions](#how-to-add-new-mpu-regions)
+- [Generic memory layout of existing MPU regions](#generic-memory-layout-of-existing-mpu-regions)
+
+## Purpose of MPU
+
+- Please refer to below ARM website link for understanding purpose of MPU:
+	- [About the MPU](http://infocenter.arm.com/help/index.jsp?topic=/com.arm.doc.ddi0439b/Chdecfea.html)
+
+## Details of MPU Configuration and Attributes
+
+- For each MPU region, the core choices are:
+	1. The start address
+	2. The size of region
+	3. Access permissions
+		1. **RO** - ReadOnly
+		2. **RW** - Read/Write
+		3. **No Access**
+	4. Execute permissions
+		1. **X** - Executable
+		2. **XN** - Execute Never
+
+- For configuring an MPU region, we need to set below MPU registers:
+	1. **MPU_RNR** : MPU Region Number register
+	2. **MPU_RBAR** : MPU Region Base Address register
+	3. **MPU_RASR** : MPU Region Attribute and Size Register
+
+- MPU_RNR is a unified identifier to the MPU system. This value should not exceed max number of MPU regions for the processor.
+- MPU_RBAR defines the starting address of a MPU region.
+- MPU_RASR encodes size of MPU region and attributes like read/write access, execute permission etc.
+
+- When configuring MPU regions, take into account that for overlapping memory ranges b/w MPU regions, attributes of highest-numbered MPU region will overwrite those set by lower MPU regions for that address range.
+
+## Prerequisites for MPU in TizenRT
+
+- In TizenRT, support for configuring MPU regions is available only for Protected Build.
+- The following configs need to be enabled for configuring MPU regions(for ARMv7 architecture):
+
+```
+	---------------------------------------------------------------------------------
+	CONFIG FLAG			VALUE		DESCRIPTION
+	---------------------------------------------------------------------------------
+	CONFIG_BUILD_PROTECTED		y		TizenRT Protected build
+	CONFIG_ARMV7M_MPU		y		MPU support
+	CONFIG_ARMV7M_MPU_NREGIONS	16		Number of MPU regions
+	---------------------------------------------------------------------------------
+```
+
+- Similarly, for other architecture versions, corresponding MPU related configs for that architecture should be enabled.
```

Comment:
MPU related configs for that architecture should be enabled, provided it supports MPU in both the cases**

Reply:
done